Source/JavaScriptCore/llint/WebAssembly.asm

```diff
@@ -640,54 +653,57 @@ end
             emit "movz x5, #0xBAD"
             emit "movz x6, #0xBAD"
             emit "movz x7, #0xBAD"
-            emit "movz x8, #0xBAD"
         end
     end
 
     clobberVolatileRegisters()
 
+
 if ARM64 or ARM64E or ARMv7
-    const JSEntrypointInterpreterCalleeSaveSpaceStackAligned = (4 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
+    const JSEntrypointInterpreterSpillSpaceStackAligned = (6 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
 else
-    const JSEntrypointInterpreterCalleeSaveSpaceStackAligned = (8 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
+    const JSEntrypointInterpreterSpillSpaceStackAligned = (8 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
 end
 
     macro saveJSEntrypointInterpreterRegisters()
-        subp JSEntrypointInterpreterCalleeSaveSpaceStackAligned, sp
+        subp JSEntrypointInterpreterSpillSpaceStackAligned, sp
         if ARM64 or ARM64E
-            storepairq boundsCheckingSize, metadataTable, -16[cfr]
-            storepairq wasmInstance, memoryBase, -32[cfr]
+            storepairq memoryBase, boundsCheckingSize, -16[cfr]
+            storepairq ws0, wasmInstance, -32[cfr]
+            # These are just for us to spill things
+            # MPCSpill
         elsif X86_64
             # These must match the wasmToJS thunk, since the unwinder won't be able to tell the difference between us and them.
             # See calleeSaveRegistersImpl
-            storep metadataTable, -0x8[cfr]
-            storep boundsCheckingSize, -0x10[cfr]
-            storep memoryBase, -0x18[cfr]
-            storep wasmInstance, -0x20[cfr]
+            storep boundsCheckingSize, -0x8[cfr]
+            storep memoryBase, -0x10[cfr]
+            storep wasmInstance, -0x18[cfr]
             # This gets restored before call too, since it doesn't get saved/restored by the wasmToJS thunk
             storep Accumulator, AccumulatorCSRSlot[cfr]
+            storep MPC, MPCCSRSlot[cfr]
             # These are just for us to spill things
             # AccumulatorSpill
-            # CP
+            # MPCSpill
+            # CalleeExecutableAddress
         else
             error
         end
     end
 
     macro restoreJSEntrypointInterpreterRegisters()
         if ARM64 or ARM64E
-            loadpairq -16[cfr], boundsCheckingSize, metadataTable
-            loadpairq -32[cfr], wasmInstance, memoryBase
+            loadpairq -16[cfr], memoryBase, boundsCheckingSize
+            loadpairq -32[cfr], ws0, wasmInstance
```

Comment:
The arm spec says operand order, unless offline asm is doing something scary. That is important too, because this needs to match whatever the JSToWasm stub does. In either case, the order should be the same, just moved up by one slot. I would not be surprised if I got the math wrong though

Reply:
Anyway, EWS caught this when I wrote the original patch, so it is probably right

Reply:
I'm not exactly sure what the ARM spec says or that it really matters (we already differ from the ARM C ABI in a variety of ways). What mostly matters that this matches what genericUnwind thinks the spill order is, which is determined by RegisterAtOffsetList. RegisterAtOffsetList fills the values in the register enum order so that's what LLInt needs to match in order to work correctly with genericUnwind.

Reply:
Ah, I misunderstood. Yes, exactly.